packages/graphql/tests/integration/issues/526.int.test.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Driver } from "neo4j-driver";
+import { graphql } from "graphql";
+import { gql } from "apollo-server";
+import neo4j from "../neo4j";
+import { Neo4jGraphQL } from "../../../src/classes";
+
+describe("https://github.com/neo4j/graphql/issues/526 - Int Argument on Custom Query Converted to Float", () => {
+    let driver: Driver;
+    let bookmarks: string[];
+    const typeDefs = gql`
+        type Movie {
+            title: String
+            tags: [Tag] @relationship(type: "HAS", direction: OUT)
+        }
+
+        type Tag {
+            name: String!
+            papers: [Movie] @relationship(type: "HAS", direction: IN)
+        }
+
+        type Query {
+            movie_tags(tagName: String = "", limit: Int): [Movie]
+                @cypher(
+                    statement: """
+                    MATCH (tag:Tag)<-[:HAS]-(movie:Movie)
+                    WHERE tag.name = $tagName
+                    RETURN movie
+                    LIMIT $limit
+                    """
+                )
+        }
+    `;
+
+    beforeAll(async () => {
+        driver = await neo4j();
+        const session = driver.session();
+
+        try {
+            await session.run(
```

Comment:
Is there a reason for creating multiple sessions within the same suite?

If we create the session in beforeAll and close it in afterAll we can avoid all the try...catch and sessions setup, may lead to slightly simpler code :thinking: .
```ts
let session: Session

beforeAll(()=>{
     session=driver.session();
})

afterAll(()=>{
     session.close()
})
```

Thoughts?

If a new session is needed per test, we could do the same in before...after

Reply:
It's good practice to create a new session per query, so I'd prefer it that way. Sessions are very light weight.